onnxruntime/core/session/inference_session.cc

```diff
@@ -869,15 +865,29 @@ common::Status InferenceSession::Load() {
 
 common::Status InferenceSession::TransformGraph(onnxruntime::Graph& graph, bool saving_model_in_ort_format) {
   // The transformer order:
-  // 1. run level 1 optimizations. these only use ONNX operators.
-  // 2. partition nodes based on EP capabilities. EPs may fuse nodes during this process.
-  // 3. run all optimizations. level 2 and 3 optimizations use contrib ops.
-  // 4. insert cast nodes
-  // 5. insert copy nodes
+  // 1. ensure potential QDQ node units have unique DQ nodes (required transformer).
```

Comment:
[```suggestion](url)
// 1. ensure potential QDQ node units have unique DQ nodes. 
  //    - This is a required transformer as the ORT code has a hard requirement there are no overlapping QDQ node units.
  //    - We run it here in case optimizers are disabled. 
```

It's good to capture the 'why' in the comments.